### ISSUE TITLE
Testing: --no-build shows SHAREDLIB_BUILD PENDING in TestStatus

### ISSUE THREAD

**Issue by samsrabin** [ESMCI/cime#4682]
In the (completed) test at `/glade/derecho/scratch/samrabin/tests_0913-1302/SMS_D_Mmpi-serial_Ld5.5x5_amazon.I1850Clm60Bgc.derecho_gnu.clm-HillslopeC.C.0913-1302/`, TestStatus shows
```
PASS SMS_D_Mmpi-serial_Ld5.5x5_amazon.I1850Clm60Bgc.derecho_gnu.clm-HillslopeC CREATE_NEWCASE
PASS SMS_D_Mmpi-serial_Ld5.5x5_amazon.I1850Clm60Bgc.derecho_gnu.clm-HillslopeC XML
PASS SMS_D_Mmpi-serial_Ld5.5x5_amazon.I1850Clm60Bgc.derecho_gnu.clm-HillslopeC SETUP
PEND SMS_D_Mmpi-serial_Ld5.5x5_amazon.I1850Clm60Bgc.derecho_gnu.clm-HillslopeC SHAREDLIB_BUILD
PASS SMS_D_Mmpi-serial_Ld5.5x5_amazon.I1850Clm60Bgc.derecho_gnu.clm-HillslopeC NLCOMP
```

even though it was set up with `--no-build`:
```bash
/glade/work/samrabin/ctsm_hs-uii/cime/scripts/create_test \
--test-id 0913-1302 \
--output-root /glade/derecho/scratch/samrabin/tests_0913-1302 \
--xml-category hillslope \
--xml-machine derecho \
--xml-compiler gnu \
--compare ctsm5.2.027 \
--baseline-root /glade/campaign/cgd/tss/ctsm_baselines \
--project P93300041 \
--retry 0 \
--namelists-only \
--no-build
```

That test used cime tag `cime6.0.246`.
EDIT: The issue still happens with `cime6.1.26`.

**Comment by jedwards4b** [comment]
So if you are planning to contribute fixes why are you using such an old cime tag?  Latest is 6.1.26

**Comment by samsrabin** [comment]
@jedwards4b I just used the version that came with one of the most recent CTSM tags (ctsm5.2.027). It still happens with `cime6.1.26`; see `/glade/derecho/scratch/samrabin/tests_0913-1302/SMS_D_Mmpi-serial_Ld5.5x5_amazon.I1850Clm60Bgc.derecho_gnu.clm-HillslopeC.C.0913-1344`.

I'm not planning to contribute a fix for this because I couldn't figure out why `SHAREDLIB_BUILD` was getting printed at all.

**Comment by samsrabin** [comment]
I'm hoping to contribute a fix to this before the CESM3 release.

**Comment by samsrabin** [comment]
This still happens as of cime6.1.59 using the following setup:
```shell
cime/scripts/create_test --no-build SMS_D_Mmpi-serial_Ld5.5x5_amazon.I1850Clm60Bgc.derecho_gnu.clm-HillslopeC
```

**Comment by samsrabin** [comment]
I tried to throw an error to pinpoint where it's getting marked as pending, but no luck. See https://github.com/ESMCI/cime/pull/4766/files

**Comment by samsrabin** [comment]
I'm still hoping to get to this.

**Comment by jgfouca** [comment]
@billsacks , yes, this is a feature that's working as intended. As you said, we cannot assume `--no-build` actually means never build.

**Comment by samsrabin** [comment]
@jgfouca:
> As you said, we cannot assume `--no-build` actually means never build.

A behavior like "Don't post a `SHAREDLIB_BUILD` status until `case.build` is actually called" would handle that elegantly, though, no? Unless—is the fear is that something might fail between the `case.build` call/job submission and the posting of `SHAREDLIB_BUILD`?

@billsacks I guess for that reason the `--never-build` option is fine, but it would need to be paired with changes that make it impossible to build the resulting case, right?
